email_notification/helpers/email_sender.py

```diff
@@ -117,8 +140,14 @@ def _send_email(self, message_builder: BaseEmailBuilder) -> None:
             from_email=settings.DEFAULT_FROM_EMAIL,
             bcc=self.recipients,
         )
-
         if html_content:
             email_message.attach_alternative(html_content, "text/html")
 
-        email_message.send(fail_silently=False)
+        if len(self.recipients) <= settings.EMAIL_MAX_RECIPIENTS:
+            email_message.send(fail_silently=False)
+        else:
+            # Send emails in batches, if there are more recipients than the maximum allowed
+            for i in range(0, len(self.recipients), settings.EMAIL_MAX_RECIPIENTS):
+                email_message_copy = copy(email_message)  # Copy the email message to avoid modifying the original
+                email_message_copy.bcc = self.recipients[i : i + settings.EMAIL_MAX_RECIPIENTS]
+                email_message_copy.send(fail_silently=False)
```

Comment:
This could be simplified using [itertools.batched](https://docs.python.org/3/library/itertools.html#itertools.batched). There is only going to be one batch when recipients length is under `EMAIL_MAX_RECIPIENTS`, so I don't think the if-else condition is needed.